test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java

```diff
@@ -150,33 +150,50 @@ public Object[][] servers() {
         };
     }
 
-    record bytes(byte[] bytes) {
+    private static class Bytes {
+        private final byte[] bytes;
+        public Bytes(byte[] bytes) {
+            this.bytes = bytes;
+        }
         @Override
         public boolean equals(Object o) {
             if (this == o) return true;
-            if (o instanceof bytes other) {
-                return Arrays.equals(bytes(), other.bytes());
+            if (o instanceof byte[]) {
+                return Arrays.equals(bytes, (byte[]) o);
+            }
+            if (o instanceof Bytes) {
+                return Arrays.equals(bytes, ((Bytes) o).getBytes());
             }
             return false;
         }
+        public byte[] getBytes() {
+            return bytes;
+        }
         @Override
-        public int hashCode() { return Arrays.hashCode(bytes()); }
+        public int hashCode() {
+            return Arrays.hashCode(bytes);
+        }
         public String toString() {
-            return "0x" + HexFormat.of()
-                    .withUpperCase()
-                    .formatHex(bytes());
+            StringBuilder builder = new StringBuilder("0x");
+            for (byte aByte : bytes) {
+                builder.append(String.format("%X", aByte).toUpperCase());
+            }
+            return builder.toString();
         }
     }
 
-    static List<bytes> ofBytes(List<byte[]> bytes) {
-        return bytes.stream().map(bytes::new).toList();
+    static List<Bytes> ofBytes(List<byte[]> bytes) {
```

Comment:
Here you could rather write `return bytes.stream().map(WebSocketProxyTest.Bytes::new).collect(Collectors.toList());` to keep as close to the original as possible.